lib/cl/server.lisp

```diff
@@ -151,62 +151,62 @@
   (:method ((location puri:uri) service)
     "Given a basic thrift uri, open a binary socket server and listen on the port."
     (let ((server (make-instance 'socket-server
-                    :socket (usocket:socket-listen (puri:uri-host location) (puri:uri-port location)
-                                                   :element-type 'unsigned-byte
-                                                   :reuseaddress t))))
+				 :socket (usocket:socket-listen (puri:uri-host location) (puri:uri-port location)
+								:element-type 'unsigned-byte
+								:reuseaddress t))))
       (unwind-protect (serve server service)
         (server-close server))))
 
-  (:method ((s socket-server) (service service))
+  (:method ((s socket-server) service)
     (loop 
-      (let ((connection (accept-connection s)))
-        (if (open-stream-p (usocket:socket-stream connection))
-          (let* ((input-transport (server-input-transport s connection))
-                 (output-transport (server-output-transport s connection))
-                 (protocol (server-protocol s input-transport output-transport)))
-            (unwind-protect (block :process-loop
-                              (handler-bind ((end-of-file (lambda (eof)
-                                                            (declare (ignore eof))
-                                                            (return-from :process-loop)))
-                                             (error (lambda (error)
-                                                      (if *debug-server*
-                                                        (break "Server error: ~s: ~a" s error)
-                                                        (warn "Server error: ~s: ~a" s error))
-                                                      (stream-write-exception protocol error)
-                                                      (return-from :process-loop))))
-                                (loop (unless (open-stream-p input-transport) (return))
-                                      (process service protocol))))
-              (close input-transport)
-              (close output-transport)))
-          ;; listening socket closed
-          (return))))))
-
-  
-(defgeneric process (service protocol)
```

Comment:
why do we replace generic function with ordinary function which can't be specialized?